standard/variables.md

```diff
@@ -287,8 +287,49 @@ if ( «expr» ) «then_stmt» else «else_stmt»
 
 For a `switch` statement *stmt* with a controlling expression *expr*:
 
-- The definite-assignment state of *v* at the beginning of *expr* is the same as the state of *v* at the beginning of *stmt*.
-- The definite-assignment state of *v* on the control flow transfer to a reachable switch block statement list is the same as the definite-assignment state of *v* at the end of *expr*.
+The definite-assignment state of *v* at the beginning of *expr* is the same as the state of *v* at the beginning of *stmt*.
+
+The definite-assignment state of *v* at the beginning of a case's guard clause is
+- If *v* is a pattern variable declared in the *switch_label*: "definitely assigned".
+- If the switch label containing that guard clause is not reachable (XREF): "definitely assigned".
+- Otherwise, the state of *v* is the same as the state of *v* after *expr*.
+
+The definite-assignment state of *v* on the control flow transfer to a reachable switch block statement list is
+- If the control transfer was due to a 'goto case' or 'goto default' statement, then the state of *v* is the same as the state at the beginning of that 'goto' statement.
+- If the control transfer was due to the `default` label of the switch, then the state of *v* is the same as the state of *v* after *expr*.
+- If the control transfer was due to an unreachable switch label, then the state of *v* is "definitely assigned".
```

Comment:
This one confuses me. I think I need an example to understand the implications of v being definitely assigned due to an unreachable switch label.

Reply:
```
bool b;
switch (1) {
  case 2 when b: // b is definitely assigned here.
    break;
}
```

Reply:
I don't think this is a common enough case to deserve an example in the text, though I admit it would help the reader make sense of the text.  This can be left to be dealt with later.

Reply:
Logged #937 to discuss later.